roles/kubernetes/master/templates/kubeadm-config.v1beta1.yaml.j2

```diff
@@ -32,6 +33,46 @@ etcd:
       caFile: {{ etcd_cert_dir }}/{{ kube_etcd_cacert_file }}
       certFile: {{ etcd_cert_dir }}/{{ kube_etcd_cert_file }}
       keyFile: {{ etcd_cert_dir }}/{{ kube_etcd_key_file }}
+{% elif etcd_mode == "kubeadm" %}
+  local:
+    imageRepository: "{{ etcd_image_repo | replace("/etcd","") }}"
```

Comment:
Please add an ability to specify an image in _kubeadm-images.yaml.j2_ as well

Reply:
Also it might be better to use _regex_replace('/etcd$','')_, as this may not work for URLs like registry.my/something/etcd/etcd

Reply:
thanks for spotting that

Reply:
I think the `$` is still missing here. I think `{{ etcd_image_repo | replace("/etcd","") }}` should be `{{ etcd_image_repo | regex_replace("/etcd$","") }}`

Reply:
@woopstar fixed